### PR TITLE
[bitnami/metrics-server] Allow specifying automountServiceAccountToken for metrics-server

### PR DESCRIPTION
**Pull request by joshuastern**
**Description of the change**

Allow for specifying the `automountServiceAccountToken` field on the metrics-server service account.

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/metrics-server/Chart.yaml [hunk 1]
```diff
@@ -23,4 +23,4 @@ name: metrics-server
 sources:
   - https://github.com/bitnami/bitnami-docker-metrics-server
   - https://github.com/kubernetes-incubator/metrics-server
-version: 5.6.0
+version: 5.6.1
```
Comment (migruiz4) [inline bitnami/metrics-server/Chart.yaml:26]: Please replace this with a minor version bump
```suggestion
version: 5.7.0
```

File: bitnami/metrics-server/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -4,4 +4,5 @@ kind: ServiceAccount
 metadata:
   name: {{ template "common.names.fullname" . }}
   labels: {{- include "common.labels.standard" . | nindent 4 }}
+automountServiceAccountToken: {{ .Values.serviceAccount.automountServiceAccountToken }}
```
Comment (migruiz4) [inline bitnami/metrics-server/templates/serviceaccount.yaml:7]:
```suggestion
{{- if .Values.serviceAccount.automountServiceAccountToken }}
automountServiceAccountToken: {{ .Values.serviceAccount.automountServiceAccountToken }}
{{- end }}
```

File: bitnami/metrics-server/values.yaml [hunk 3]
```diff
@@ -61,6 +61,10 @@ serviceAccount:
   ## The name of the ServiceAccount to use.
   ## If not set and create is true, a name is generated using the common.names.fullname template
   # name:
+  ## Automount API credentials for a service account.
+  ##
+  automountServiceAccountToken: true
```
Comment (migruiz4) [inline bitnami/metrics-server/values.yaml:66]:
```suggestion
  # automountServiceAccountToken: true
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @joshuastern ,

Thank you for your contribution! I have a change request, according to [Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/), `automountServiceAccountToken` is available in Kubernetes 1.6+, and if omitted it will default to `true`.

To prevent version issues I think it would be better to include `automountServiceAccountToken` as an optional setting (please check suggestions below).

File: bitnami/metrics-server/Chart.yaml (hunk 1 above)
Comment (joshuastern) [reply bitnami/metrics-server/Chart.yaml]: 👍

File: bitnami/metrics-server/values.yaml (hunk 3 above)
Comment (joshuastern) [reply bitnami/metrics-server/values.yaml]: 👍

**Review by migruiz4 — APPROVED** [review]
Looks good to me, thank you very much for contributing and applying the suggested changes!

File: bitnami/metrics-server/templates/serviceaccount.yaml [hunk 4]
```diff
@@ -4,4 +4,7 @@ kind: ServiceAccount
 metadata:
   name: {{ template "common.names.fullname" . }}
   labels: {{- include "common.labels.standard" . | nindent 4 }}
+{{- if .Values.serviceAccount.automountServiceAccountToken }}
```
Comment (mkilchhofer) [inline bitnami/metrics-server/templates/serviceaccount.yaml:7]: Hi @migruiz4 and @joshuastern,
This change does not make sense anymore ;-)
When you provide `false` in the values, this conditional does not render the `false`

Comment (migruiz4) [reply bitnami/metrics-server/templates/serviceaccount.yaml:7]: Hi @mkilchhofer,

My bad, thank you for raising this issue.

**Pull request merged**